controller/root.go

```diff
@@ -0,0 +1,178 @@
+	"github.com/fabric8-services/fabric8-wit/errors"
+	"github.com/fabric8-services/fabric8-wit/jsonapi"
+	"github.com/fabric8-services/fabric8-wit/log"
+	"github.com/fabric8-services/fabric8-wit/rest"
+	"github.com/fabric8-services/fabric8-wit/swagger"
+	"github.com/goadesign/goa"
+	"github.com/satori/go.uuid"
+	"net/http"
+	"strings"
+)
+
+type asseter interface {
+	Asset(name string) ([]byte, error)
+}
+
+// RootController implements the root resource.
+type RootController struct {
+	*goa.Controller
+	FileHandler asseter
+}
+
+type workingFileFetcher struct{}
+
+func (s workingFileFetcher) Asset(fileName string) ([]byte, error) {
+	return swagger.Asset(fileName)
+}
+
+var _ asseter = workingFileFetcher{}
+var _ asseter = (*workingFileFetcher)(nil)
+
+// NewRootController creates a root controller.
+func NewRootController(service *goa.Service) *RootController {
+	return &RootController{
+		Controller:  service.NewController("RootController"),
+		FileHandler: workingFileFetcher{},
+	}
+}
+
+// List runs the list action.
+func (c *RootController) List(ctx *app.ListRootContext) error {
+	roots, err := getRoot(ctx.Request, c.FileHandler)
+	if err != nil || roots == nil {
+		return jsonapi.JSONErrorResponse(
+			ctx, err)
+	}
+
+	res := &app.RootSingle{}
+	res.Data = roots
+	return ctx.OK(res)
+}
+
+// Get a list of all endpoints formatted to json api format.
+func getRoot(request *http.Request, fileHandler asseter) (*app.Root, error) {
+	// Get an unmarshal swagger specification
+	result, err := getUnmarshalledSwagger(fileHandler)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get and iterate over paths from swagger specification.
+	swaggerPaths, err := getSwaggerFieldAsMap("paths", result)
+	if err != nil {
+		return nil, err
+	}
+
+	namedPaths := make(map[string]interface{})
+	for path, swaggerPath := range swaggerPaths {
+
+		// Currently not supporting endpoints that contain parameters.
+		if !strings.Contains(path, "{") {
+
+			// Use the segments in the path to construct a meaningful name to use for the path.
+			key := strings.Replace(path, "/", "_", -1)
+			key = strings.Replace(key, "_", "", 1)
+
+			pathsObj, err := swaggerPath.(map[string]interface{})
+			if !err {
+				return nil, errorLogAndReturn("Unable to assert correct format for swagger specifiation")
+			}
+
+			// Get the x-tag value. If the tag exists, use it as path name.
+			xtag, _ := getSwaggerFieldAsString("x-tag", pathsObj, false)
+			if len(xtag) > 0 {
+				key = xtag
+			}
+
+			// Set the related field and link objects for each name.
+			namedPaths[key] = map[string]interface{}{
+				"links": map[string]string{
+					"related": path,
+				},
+			}
+		}
+	}
+
+	id := uuid.NewV4()
+	basePath, err := getSwaggerFieldAsString("basePath", result, true)
+	if err != nil {
+		return nil, err
+	}
+
+	basePath = rest.AbsoluteURL(request, basePath)
+	links := &app.GenericLinksForRoot{
+		Self: &basePath,
+	}
+	return &app.Root{Relationships: namedPaths, ID: &id, Links: links}, nil
```

Comment:
We have a `ptr` package to get pointer for passed in values. For each value type (e.g. string, int, ...) there's a function inside that package. Since you don't use the `id` variable here more than once you can use that package like so:

```go
return &app.Root{Relationships: namedPaths, ID: ptr.UUID(uuid.NewV4()), Links: links}, nil
```